includes/faveo-pricing-functions.php

```diff
@@ -1,400 +1,260 @@
 <?php
+// Detect user IP
+function fhai_get_user_ip() {
+    $ip = '';
+
+    if ( isset( $_SERVER['HTTP_CLIENT_IP'] ) && ! empty( $_SERVER['HTTP_CLIENT_IP'] ) ) {
+        $ip = sanitize_text_field( wp_unslash( $_SERVER['HTTP_CLIENT_IP'] ) );
+    } elseif ( isset( $_SERVER['HTTP_X_FORWARDED_FOR'] ) && ! empty( $_SERVER['HTTP_X_FORWARDED_FOR'] ) ) {
+        $ipList = explode( ',', sanitize_text_field( wp_unslash( $_SERVER['HTTP_X_FORWARDED_FOR'] ) ) );
+        $ip     = trim( $ipList[0] );
+    } elseif ( isset( $_SERVER['REMOTE_ADDR'] ) && ! empty( $_SERVER['REMOTE_ADDR'] ) ) {
+        $ip = sanitize_text_field( wp_unslash( $_SERVER['REMOTE_ADDR'] ) );
+    }
+
+    return $ip;
+}
+
+
+// Get country from IP using ipapi.co (secure HTTPS)
+function fhai_get_country_from_ip($ip) {
+    // Fallback for localhost testing
+    if ($ip === '127.0.0.1' || $ip === '::1') {
+        return 'US';
+    }
+
+    $response = wp_remote_get("https://ipapi.co/{$ip}/json/");
+    if (is_wp_error($response)) {
+        return 'US'; // fallback
+    }
+
+    $data = json_decode(wp_remote_retrieve_body($response), true);
+
+    // ipapi.co returns "country" instead of "countryCode"
+    return !empty($data['country']) ? strtoupper($data['country']) : 'US';
+}
+
+// Currency symbol mapping by code
+function fhai_currency_symbol_by_code($currency_code) {
+    $map = array(
+        'USD' => '$', 'INR' => '₹', 'GBP' => '£', 'EUR' => '€',
+        'CAD' => 'C$', 'AUD' => 'A$', 'NZD' => 'NZ$', 'JPY' => '¥',
+        'CNY' => '¥', 'SGD' => 'S$', 'HKD' => 'HK$', 'KRW' => '₩',
+        'RUB' => '₽', 'BRL' => 'R$', 'ZAR' => 'R', 'MXN' => 'MX$',
+        'CHF' => 'CHF', 'SEK' => 'kr', 'NOK' => 'kr', 'DKK' => 'kr',
+        'PLN' => 'zł', 'CZK' => 'Kč', 'HUF' => 'Ft', 'TRY' => '₺',
+        'SAR' => '﷼', 'AED' => 'د.إ', 'ILS' => '₪', 'THB' => '฿',
+        'IDR' => 'Rp', 'MYR' => 'RM', 'PHP' => '₱', 'VND' => '₫',
+        'NGN' => '₦'
+    );
+    return isset($map[$currency_code]) ? $map[$currency_code] : '$';
```

Comment:
Instead of isset null collasing can be added, if we know it's going to be null. isset will return true for empty string with that key